gadget/validate.go

```diff
@@ -35,6 +35,45 @@ type validationState struct {
 	SystemSave *VolumeStructure
 }
 
+type ValidationConstraints struct {
```

Comment:
a doc-comment on this might be nice to explain what functions this is used with

Reply:
I'm adding:

```
// ValidationConstraints carries extra constraints on top of those
// implied by the model to use for gadget validation.
// They might be constraints that are determined only at runtime.
```